### PR TITLE
Feature/favourite

### PR DESCRIPTION
**Pull request by aagavin**
# favourite

## This PR includes
* CRUD For favourites
* Rewrite of all features needed from the broken ID
* Frontend update

This closes issue #10, #11 and #16

### CODE REVIEW

File: apitanic/controller/imdb.py [hunk 1]
```diff
@@ -1,41 +1,72 @@
...
-    return json({'data': {'movies': imdb.popular_movies()}})
+    popular_url = 'http://www.imdb.com/chart/moviemeter'
+    data = requests.get(popular_url).text
+    raw_string = lxml.html.fromstring(data)
+
+    posters = raw_string.xpath('//td[@class=\'posterColumn\']/a')
+    titles = raw_string.xpath('//td[@class=\'titleColumn\']/a')
+    ratings = raw_string.xpath('//td[@class=\'ratingColumn imdbRating\']')
+
+    popular_movies_list: list = []
```
Comment (aramrattan) [inline apitanic/controller/imdb.py]: Python 3.5 does not support variable annotations

File: apitanic/model/firebase.py [hunk 2]
```diff
@@ -22,7 +23,10 @@
 class Firebase:
 
     def __init__(self):
-        pass
+        self.favourites_ref: CollectionReference = firebase_db.collection('favourites')
```
Comment (aramrattan) [inline apitanic/model/firebase.py]: variable annotation

**Review by aramrattan — CHANGES_REQUESTED** [review]
There are a few Python 3.6 features in here that are not compatible with 3.5

**Comment by aagavin** [comment]
Fixed

**Review by aramrattan — APPROVED** [review]

**Pull request merged**